test/server/cards/02_SHD/units/FrontierTrader.spec.ts

```diff
@@ -0,0 +1,36 @@
+import { ZoneName } from '../../../../../server/game/core/Constants';
+
+describe('Frontier Trader', function() {
+    integration(function(contextRef) {
+        describe('Frontier Trader\'s ability', function() {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['frontier-trader'],
+                        deck: ['wampa', 'battlefield-marine', 'atst', 'atst'],
+                        resources: ['armed-to-the-teeth',
+                            'covert-strength',
+                            'chewbacca#pykesbane',
+                            'battlefield-marine', 'atst', 'atst', 'atst', 'atst', 'atst', 'atst',
+                            'collections-starhopper',
+                            'atst', 'atst', 'atst', 'atst', 'atst', 'atst', 'atst'
+                        ],
+                        base: 'chopper-base'
+                    }
+                });
+            });
+
+            it('should be able to take resource back to hand and replace it with top card from deck', function () {
+                const { context } = contextRef;
+
+                context.player1.clickCard(context.frontierTrader);
+                expect(context.player1).toHavePassAbilityButton();
+                context.player1.clickCard(context.collectionsStarhopper);
```

Comment:
Any time we are creating a `targetResolver` we need to have at least one instance of `toBeAbleToSelectExactly()` to confirm that the targeting filters are set up correctly (in this case there's some concern that it might be able to target opponent's resources)